fecgov/fec-cms
# Feature/5713 upgrade django v4.2

## Summary (required)

- Resolves #5713 

This ticket upgrades Django to version 4.2. I'll include a list of changes and links to the documentation in a comment. 

### Required reviewers 2 - 3 developers 


## Impacted areas of the application

General components of the application that this PR will affect:

-  Django settings 
- home page banner announcements 

## How to test
I recommend testing on dev or stage instead of locally. 

stage/dev:
1. Deploy test branch to dev or stage 
2. Export data table to test post request from trusted origin 
3. Check that static images work 
4. Test creating home page banner announcement 
5. Test accessing files located in S3 bucket

locally: 
1. Create new virtualenv 
2. `pyenv activate <new virtualenv>`
3. `pip install -r requirements.txt`
4. `npm install`
6. `npm run build`
7. `pytest`
8. `cd fec/`
9. `./manage.py runserver`
10. Export data table to test post request from trusted origin 
11. Check that static images work 
12. Test creating home page banner announcement
